pkg/controller/egress_router/egress_router_controller.go

```diff
@@ -173,13 +173,14 @@ func (r *EgressRouterReconciler) ensureEgressRouter(manifestDir string, namespac
 	data := render.MakeRenderData()
 	data.Data["ReleaseVersion"] = os.Getenv("RELEASE_VERSION")
 	data.Data["EgressRouterNamespace"] = namespace
-	if isItValidCidr(string(router.Spec.Addresses[0].IP)) {
+	if isItValidIPAddress(string(router.Spec.Addresses[0].IP)) {
```

Comment:
@msherif1234 I have zero context here, just doing a drive-by. What sets the EgressRouter's Spec.Addresses array? If it used to be a CIDR, I don't think we can just change it over to a plain IP without breaking existing EgressRouter objects, right?

If that's true, at the very least we should warn if we find a CIDR, then ParseCIDR() it to pull off the IP and then just use that with the /24.

Reply:
@dcbw me too :) but thanks for driving by!! 
its not CIDR its just array of strings "ips" I am with u we should let customer configure the prefix length and not hardcode it seems like a kludge I will see if @tssurya and @squeed  ok with me changing the API